scripts/schema/subset_filter.py

```diff
@@ -0,0 +1,77 @@
+import glob
+import yaml
+
+# This script takes all ECS and custom fields already loaded, and lets users
+# filter ou
+
+
+def filter(fields, subset_file_globs):
+    '''
+    Takes the deeply nested field structure and the subset file names.
+
+    It returns a copy of the fields that matches the whitelist defined in the subset.
+    '''
+    if not subset_file_globs or subset_file_globs == []:
+        return fields
+    subset_definitions = load_subset_definitions(subset_file_globs)
+    filtered_fields = extract_matching_fields(fields, subset_definitions)
+    return filtered_fields
+
+
+def load_subset_definitions(file_globs):
+    subsets = {}
+    for f in eval_globs(file_globs):
+        raw = load_yaml_file(f)
+        merge_subsets(subsets, raw)
+    if not subsets:
+        raise ValueError('--subset specified, but no subsets found in {}'.format(file_globs))
+    return subsets
+
+
+def load_yaml_file(f):
+    return yaml.safe_load(f.read())
```

Comment:
When playing with the branch locally i found that `f` here is a string- need a `with open(f) as ...`

Reply:
The joy of relying too much on unit tests. Thanks for catching this!

We'll add CLI tests to catch those ;-)